packages/components/progress/src/ProgressLabel.tsx

```diff
@@ -0,0 +1,29 @@
+import { useId } from '@radix-ui/react-id'
+import { useMergeRefs } from '@spark-ui/use-merge-refs'
+import { ComponentPropsWithoutRef, forwardRef, useCallback } from 'react'
+
+import { useProgress } from './ProgressContext'
+
+export type ProgressLabelProps = ComponentPropsWithoutRef<'span'>
+
+export const ProgressLabel = forwardRef<HTMLSpanElement, ProgressLabelProps>(
+  ({ id: idProp, children, ...others }, forwardedRef) => {
+    const id = useId(idProp)
+    const { onLabelId } = useProgress()
+    const rootRef = useCallback(
+      (el: HTMLSpanElement) => {
+        onLabelId(el ? id : undefined)
```

Comment:
I'm not sure I understand what's going on here, we have some `React.Context`that exposes a `onLabelId` function,
and you're calling that function here and passing it the generated id from the line above. But I'm not seeing where we're actually setting the id on the element itself 🤔

Reply:
Yes, I remember what you said yesterday about SSR and how React Spectrum was dealing with it, but I overlooked the `onLabelId: setLabelId` part 👍